## Statistical Data Analysis

In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [6]:
nba_data = pd.read_csv('NBA Stats 202324 All Stats.csv')
nba_data.head()

RANK                     NAME TEAM POS   AGE  GP   MPG  USG%   TO%  FTA  \
0     1              Joel Embiid  Phi   C  30.2   6  41.4  35.7  15.8   78   
1     2            Jalen Brunson  Nyk   G  27.8  13  39.8  36.4   9.3  120   
2     3           Damian Lillard  Mil   G  33.9   4  39.1  31.4  10.0   38   
3     4  Shai Gilgeous-Alexander  Okc   G  25.9  10  39.9  32.3   8.9   81   
4     5             Tyrese Maxey  Phi   G  23.6   6  44.6  28.1   8.6   28   

   ...  APG  SPG  BPG  TPG   P+R   P+A  P+R+A    VI   ORtg   DRtg  
0  ...  5.7  1.2  1.5  4.2  43.8  38.7   49.5  12.2  117.1  108.0  
1  ...  7.5  0.8  0.2  2.7  35.7  39.8   43.2   9.3  114.8  114.7  
2  ...  5.0  1.0  0.0  2.3  34.5  36.3   39.5   8.2  127.6  115.7  
3  ...  6.4  1.3  1.7  2.2  37.4  36.6   43.8  11.2  118.3  106.9  
4  ...  6.8  0.8  0.3  2.2  35.0  36.7   41.8   9.1  120.9  113.3  

[5 rows x 29 columns]

## <span style="color:blue">  1. Descriptive Statistics </span>

• foundation of any analysis & essential for understanding the distribution of your data 

In [16]:
# Displaying descriptive statistics
summary_stats = nba_data.describe()
print(summary_stats)

             RANK         AGE          GP         MPG        USG%         TO%  \
count  213.000000  213.000000  213.000000  213.000000  213.000000  213.000000   
mean   107.000000   27.738967    7.657277   19.994366   17.933803   12.603756   
std     61.631972    4.573337    5.123927   13.518864    8.838268   12.279127   
min      1.000000   20.300000    1.000000    0.900000    0.000000    0.000000   
25%     54.000000   24.200000    4.000000    6.000000   12.600000    6.600000   
50%    107.000000   27.000000    6.000000   19.100000   17.000000   11.000000   
75%    160.000000   31.100000   11.000000   33.600000   22.200000   14.600000   
max    213.000000   39.500000   20.000000   44.600000   85.000000  100.000000   

              FTA         FT%         2PA         2P%  ...         APG  \
count  213.000000  213.000000  213.000000  213.000000  ...  213.000000   
mean    15.347418    0.577742   38.403756    0.470427  ...    1.818310   
std     23.407817    0.367595   52.976111    0.2

### <span style="color:red">  Results:</span> 

### General Player Info

**RANK:** Players are ranked from 1 to 213.

**AGE:**

• Mean age: 27.7 years

• Range: 20 to 39.5 years → <span style="color:red; font-size: larger;">  indicating a mix of young talent & veterans.</span>

**GP (Games Played):**

• Mean: 7.7 games

• Max: 20 games

• Many players have played fewer than 10 games → <span style="color:red; font-size: larger;">  suggesting either a shortened season, injuries, or limited participation.</span>


### Playing Time & Role

**MPG (Minutes Per Game):**

• Mean: 19.99 minutes

• Wide range (min: 0.3, max: 44.6) → <span style="color:red; font-size: larger;">  showing clear role differences (bench vs starters).</span>

**USG% (Usage Rate):**

• Mean: 17.93%.

• High standard deviation (~8.84) → <span style="color:red; font-size: larger;">  suggesting players vary widely in ball usage.</span>

**TO% (Turnover Rate):**

• Average: 12.66%.

• Goes up to 100% →  <span style="color:red; font-size: larger;">  may indicate data anomaly or very low-minute players with many turnovers.</span>



### Shooting & Scoring

**FTA (Free Throw Attempts):**

• Mean: 15.35 attempts across games (not per game).

• Max: 141 → <span style="color:red; font-size: larger;">  suggests some highly aggressive scorers.</span>

**FT% (Free Throw %):**

• Average: 57.8% (low compared to typical NBA averages).

→ <span style="color:red; font-size: larger;">  Indicates poor overall free throw performance.</span>

**2PA & 2P% (Two-point attempts & %):**

• Mean 2PA: 38.4 attempts total.

• 2P%: 47% → <span style="color:red; font-size: larger;">  which is average.</span>

### Passing & Assists

**APG (Assists Per Game):**

• Mean: 1.82 assists

• Max: 8.8 → <span style="color:red; font-size: larger;">shows a few elite playmakers</span>

### Defense

• **SPG (Steals Per Game):** 0.54 average

• **BPG (Blocks Per Game):** 0.39 average

→ <span style="color:red; font-size: larger;">  Indicates limited defensive stats from the general player pool.</span>

### Turnovers & Combined Stats

• **TPG (Turnovers Per Game):** 0.97

• **P+R, P+A, P+R+A:**  (represent combined Points + Rebounds, Points + Assists, etc.)

• **P+R+A max = 50.8** → <span style="color:red; font-size: larger;">  indicating a superstar-level stat line.</span>

### Advanced Metrics

**VI (Value Index?):**

• Mean: 5.68.

• High performer at 15.

**ORtg (Offensive Rating) & DRtg (Defensive Rating):**

• Mean ORtg: 94.6.

• Mean DRtg: 93.17.

• Max ORtg (300) and DRtg (122) are unusually high → <span style="color:red; font-size: larger;">  could suggest outliers or calculation artifacts.</span>

